### PR TITLE
[chore] delete from user_repo_permissions in DeleteAllUserPermissions

### PR DESCRIPTION
**Pull request by kopancek**
## Description

Modifies the method `DeleteAllUserPermissions` of perms_store.go to delete from the new unified user_repo_permissions table as well as from the legacy table.

## Test plan

unit test modified

### CODE REVIEW

File: enterprise/internal/database/perms_store.go [hunk 1]
```diff
@@ -1426,6 +1426,10 @@ func (s *permsStore) DeleteAllUserPermissions(ctx context.Context, userID int32)
 	ctx, save := s.observe(ctx, "DeleteAllUserPermissions", "")
 	defer func() { save(&err, otlog.Int32("userID", userID)) }()
 
+	// first delete from the unified table
+	if err = s.execute(ctx, sqlf.Sprintf(`DELETE FROM user_repo_permissions WHERE user_id = %d`, userID)); err != nil {
+		return errors.Wrap(err, "execute delete user repo permissions query")
+	}
```
Comment (mrnugget) [inline enterprise/internal/database/perms_store.go:1432]: Don't we want a transaction here in this method to ensure both queries succeed?

File: enterprise/internal/database/perms_store_test.go [hunk 2]
```diff
@@ -2611,59 +2611,111 @@ func testPermsStore_SetPendingPermissionsAfterGrant(db database.DB) func(*testin
...
+	}); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := s.SetRepoPermissions(ctx, &authz.RepoPermissions{
+		RepoID:  2,
+		Perm:    authz.Read,
+		UserIDs: toMapset(1, 2),
+	}); err != nil {
+		t.Fatal(err)
+	}
```
Comment (mrnugget) [inline enterprise/internal/database/perms_store_test.go:2666]: Use the loop, luke

```go
for _, repoID := range []api.RepoID{1, 2} {
}
```

**Review by mrnugget — APPROVED** [review]
Approve but wondering about the transaction

File: enterprise/internal/database/perms_store.go (hunk 1 above)
Comment (kopancek) [reply enterprise/internal/database/perms_store.go:1432]: Yes, we do. Good point.

**Pull request merged**